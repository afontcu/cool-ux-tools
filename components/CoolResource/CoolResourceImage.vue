<template>
  <vue-clazy-load :src="src" class="CoolImage">
    <transition name="fade">
      <img :src="src" :alt="alt" class="CoolImage__img">
    </transition>
    <transition slot="placeholder" name="fade">
      <div class="CoolImage__placeholder"/>
    </transition>
  </vue-clazy-load>
</template>

<script>
import { VueClazyLoad } from 'vue-clazy-load'

export default {
  name: 'CoolResourceImage',
  components: {
    VueClazyLoad,
  },
  props: {
    src: { type: String, required: true },
    alt: { type: String, required: true },
  },
}
</script>

<style lang="scss">
.CoolImage {
  --image-size: 75px;
}
@media screen and (min-width: $small-breakpoint) {
  .CoolImage {
    --image-size: 150px;
  }
}

.CoolImage__img {
  width: calc(var(--image-size) + 1vw);
  transition: transform $global-transition ease;
  border-radius: $global-radius;

  &:hover {
    transform: rotate3d(0, 0, 1, -2deg) scale(1.1);
  }
}

.CoolImage__placeholder {
  width: var(--image-size);
  height: var(--image-size);
  background-color: #f1f1f2;
}

.fade-enter,
.fade-leave-to {
  opacity: 0;
  transform: rotate3d(0, 0, 1, 2deg) scale(0.9);
}
</style>
