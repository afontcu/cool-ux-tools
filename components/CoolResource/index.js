import CoolResource from './CoolResource'

export default CoolResource
