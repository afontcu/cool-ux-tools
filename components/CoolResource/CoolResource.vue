<template>
  <div :class="[ classes ]" class="Cool-resource u-padding-small o-media o-media--top">
    <a :href="link"
       :title="name"
       target="_blank"
       class="Cool-resource__image-link o-media__img u-flex u-flex-column u-flex-justify-center u-padding-bottom">
      <cool-resource-image :src="imageSrc" :alt="name" />
    </a>
    <div class="o-media__body u-flex u-flex-column">
      <h3 class="u-text-special-font u-text-h2 u-text-uppercase">
        {{ name }}
      </h3>
      <span v-if="description" class="u-flex-default u-margin-bottom-small">
        {{ description }}
      </span>
      <fancy-link :href="link" :title="name" target="_blank">
        Go to {{ name }} &rarr;
      </fancy-link>
    </div>
  </div>
</template>

<script>
import FancyLink from '@/components/FancyLink'
import descriptions from '~/data/descr.json'
import CoolResourceImage from './CoolResourceImage'

export default {
  name: 'CoolResource',
  components: {
    FancyLink,
    CoolResourceImage,
  },
  props: {
    category: { required: true, type: String },
    link: { required: true, type: String },
    name: { required: true, type: String },
    image: { required: true, type: String },
  },
  data () {
    return {
      descriptions,
    }
  },
  computed: {
    description () {
      return this.descriptions.find(d => d.link === this.link).description
    },
    imageSrc () {
      return `/logos${this.image}`
    },
    classes () {
      const className = this.name.toLowerCase().replace(/[\W_]+/g, '')
      return [`Resource--${className}`]
    },
  },
}
</script>
