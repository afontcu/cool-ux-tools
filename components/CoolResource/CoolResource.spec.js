import CoolResource from './index.js'
import { mount } from '@vue/test-utils'
require('intersection-observer')

const propsData = {
  category: 'category',
  link: 'link.com',
  name: 'Name',
  image: 'image.png',
}

describe('CoolResource.vue', () => {
  it('is a component', () => {
    const wrapper = mount(CoolResource, { propsData })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(CoolResource, { propsData })
    expect(wrapper.classes()).toContain('Cool-resource')
  })

  it('renders a snapshot', () => {
    const wrapper = mount(CoolResource, { propsData })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
