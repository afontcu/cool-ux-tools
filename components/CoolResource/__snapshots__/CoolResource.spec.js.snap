// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoolResource.vue renders a snapshot: default 1`] = `

<div class="Cool-resource u-padding-small o-media o-media--top Resource--name">
  <a href="link.com"
     title="Name"
     target="_blank"
     rel="noopener"
     class="Cool-resource__image-link o-media__img u-flex u-flex-column u-flex-justify-center u-padding-bottom"
  >
    <div class="loading">
      <div class="Cool-resource-image__placeholder"
           name="fade"
      >
      </div>
    </div>
  </a>
  <div class="o-media__body u-flex u-flex-column">
    <h3 class="u-text-special-font u-text-h2 u-text-uppercase">
      Name
    </h3>
    <a href="link.com"
       title="Name"
       class="FancyLink"
       target="_blank"
       rel="noopener"
    >
      Go to Name â†’
    </a>
  </div>
</div>

`;
